### PR TITLE
[DOC] Update docs for 23.10.0 release [skip ci]

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR updates download docs for 23.10.0 release.

Signed-off-by: Suraj Aralihalli <suraj.ara16@gmail.com>

### CODE REVIEW

File: docs/archive.md [hunk 1]
```diff
@@ -5,6 +5,85 @@ nav_order: 15
...
+  and [RAPIDS Accelerator for Apache Spark 23.08.2 jars.asc](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/23.08.2/rapids-4-spark_2.12-23.08.2.jar.asc)
+* Download the [PUB_KEY](https://keys.openpgp.org/search?q=sw-spark@nvidia.com).
+* Import the public key: `gpg --import PUB_KEY`
+* Verify the signature: `gpg --verify rapids-4-spark_2.12-23.08.2.jar.asc rapids-4-spark_2.12-23.08.2.jar`
+
+The output if signature verify:
+
+	gpg: Good signature from "NVIDIA Spark (For the signature of spark-rapids release jars) <sw-spark@nvidia.com>"
+
+### Release Notes
```
[the remark this answers is not in the document]
Comment (SurajAralihalli) [reply docs/archive.md:71]: Yeah, 23.08.2 release has similar release notes as 23.08.1.
I have updated the release notes for 23.10.0 in download.md

File: docs/download.md [hunk 2]
```diff
@@ -61,38 +60,40 @@ for your hardware's minimum driver version.
 *For Cloudera and EMR support, please refer to the
 [Distributions](https://docs.nvidia.com/spark-rapids/user-guide/latest/faq.html#which-distributions-are-supported) section of the FAQ.
 
-### Download v23.08.2
+#### RAPIDS Accelerator's Support Policy for Apache Spark
```
Comment (SurajAralihalli) [inline docs/download.md:63]: @mattf this section is for the support policy.
Please suggest an alternate if this description doesn't accurately convey our policy.

**Comment by firestarman** [comment]
LGTM

**Review by sameerz — APPROVED** [review]

**Comment by pxLi** [comment]
build

**Pull request merged**